snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -328,8 +328,15 @@ def _ensure_compatible_build_environment(self) -> None:
     def _install_file(self, *, path: str, content: bytes, permissions: str) -> None:
         basename = os.path.basename(path)
 
-        with tempfile.NamedTemporaryFile(suffix=basename) as temp_file:
-            temp_file.write(content)
+        # Push to a location that can be written to by all backends
+        # with unique files depending on path.
+        # The path should be a valid path for the target.
+        remote_file = "/var/tmp/{}".format(base64.b64encode(path.encode()).decode())
+
+        # Windows cannot open the same file twice, so write to a temporary file that
+        # would later be deleted manually.
+        with tempfile.NamedTemporaryFile(delete=False, suffix=basename) as temp_file:
+            temp_file.write(content.encode())
```

Comment:
```suggestion
            temp_file.write(content)
```

Reply:
Ah yes, I messed up that conflict, thanks